src/Compute/Compute/Extension/SqlServer/SetAzureVMSqlServerExtensionCommand.cs

```diff
@@ -136,11 +139,23 @@ public override void ExecuteCmdlet()
             {
                 try
                 {
-                    op = VirtualMachineExtensionClient.CreateOrUpdateWithHttpMessagesAsync(
-                        ResourceGroupName,
-                        VMName,
-                        Name ?? VirtualMachineSqlServerExtensionContext.ExtensionPublishedName,
-                        parameters).GetAwaiter().GetResult();
+                    if (NoWait.IsPresent)
+                    {
+                        op = VirtualMachineExtensionClient.BeginCreateOrUpdateWithHttpMessagesAsync(
+                            ResourceGroupName,
+                            VMName,
+                            Name ?? VirtualMachineSqlServerExtensionContext.ExtensionPublishedName,
+                            parameters).GetAwaiter().GetResult();
+                    }
+                    else
+                    {
+                        op = VirtualMachineExtensionClient.CreateOrUpdateWithHttpMessagesAsync(
+                            ResourceGroupName,
+                            VMName,
+                            Name ?? VirtualMachineSqlServerExtensionContext.ExtensionPublishedName,
+                            parameters).GetAwaiter().GetResult();
+                    }
+                    
                     break;
                 }
```

Comment:
I am not sure this deserialization will work on the try/catch - I would remove this one for now or just do the try/catch around the actual long-runnign operation

Reply:
Ok. Done